### PR TITLE
RFC-10 Device Locking 

### PR DESCRIPTION
**Pull request by piogor**
PR with clean history, replacing #109

### CODE REVIEW

**Comment by piogor** [comment]
@hardion, I did cherry-picking, so it should be in-line with your original PR and credits your contribution but please check it :-).

File: 10/RequestReply.md [hunk 1]
```diff
@@ -127,6 +127,63 @@ Operations case:
...
 NOTE All GROUPED REQUESTS should be logged once with the same information as the unique request
 
+### Device Locking
+A Client has the possibility to avoid any modification of a Device from other Clients. This system is so-called Device Lock. The Device Server has here the role and the responsibility to apply this mechanism.
+
+The following specifies the Device Lock system:  
+* Only one Device Lock per Device SHALL be activated.
+* A Device Lock SHALL be only owned by one Client.
+
+When a Client activates a Device Lock, the Server MUST reject any of the following request from other Client:
```
Comment (bourtemb) [inline 10/RequestReply.md]:
```suggestion
When a Client activates a Device Lock, the Server MUST reject any of the following requests from any other Client:
```

File: 10/RequestReply.md [hunk 2]
```diff
@@ -127,6 +127,63 @@ Operations case:
...
+* Command call except for State Command, Status Command and all commands listed in the Allowed Commands list (described further down).
+* Write Attribute Request (including the atomic Write Read request)
+* Write Pipe Request (including the atomic Write Read request)
+* Setting Attribute Configuration
+* Setting Pipe Configuration
+* Setting Polling Configuration
+* Setting Logging Configuration
+In general it is RECOMMANDED to block any request resulting a mutation of a Device.
```
Comment (bourtemb) [inline 10/RequestReply.md]:
```suggestion
In general it is RECOMMENDED to block any request resulting in a mutation of a Device.
```

File: 10/RequestReply.md [hunk 3]
```diff
@@ -127,6 +127,63 @@ Operations case:
...
+
+An Allowed Command is not affected by any the Device Lock, even this implies a modification of the Device state. The DeviceServer define a list of Allowed Command per Device Class which can only be changed at the Initialisation phase (TODO find the exact term) of the Device Server.
```
Comment (bourtemb) [inline 10/RequestReply.md]:
```suggestion
An Allowed Command is not affected by the Device Lock, even if this command execution implies a modification of the Device state. The Device Server defines a list of Allowed Commands per Device Class which can only be changed at the Initialisation phase (TODO find the exact term) of the Device Server.
```

File: 10/RequestReply.md [hunk 4]
```diff
@@ -127,6 +127,63 @@ Operations case:
...
+TODO: Explain how to set/get this list?
+
+Other Clients have the possibility to check a Device Lock is currently activated (TODO See DeviceServer command)
+
+The request of a Device Lock is done throught the DeviceServer command (?). The protocol of activation SHALL be the responsibility of the Device Server, following this sequence: (?)
```
Comment (bourtemb) [inline 10/RequestReply.md]:
```suggestion
The request of a Device Lock is done through the Device Server command (?). The protocol of activation SHALL be the responsibility of the Device Server, following this sequence: (?)
```

File: 10/RequestReply.md [hunk 5]
```diff
@@ -127,6 +127,63 @@ Operations case:
...
+* a counter (?) representing the number of valid activation from the same Client 
+* the activation time in second defining the locking period from the last activation timestamp  # time_t in the implementation?)
+
+The desactivation  of a Device Lock is defined by these conditions:
+* Time of activation expired
+* The Connection to the Client is lost
+* The Device Server is reinitialised 
+* ... TODO
+
+A Client Identication SHOULD cary these information:
```
Comment (bourtemb) [inline 10/RequestReply.md]:
```suggestion
A Client Identification SHOULD carry these information:
```

File: 10/RequestReply.md [hunk 6]
```diff
@@ -127,6 +127,62 @@ Operations case:
...
+* The Client which locked the Devcie call DeviceServer command UnLockDevice for the locked Device,
+* or Time of activation expired,
+* or the Connection to the Client is lost,
+* The Device Server is reinitialised.
+
+A Client Identication SHOULD cary these information:
+* Hostname where the Client process runs
+* PID of the Client process
+* IP of the Client connection
+It is recommended to include any other informations which may help the administrators of the system.
```
Comment (piogor) [inline 10/RequestReply.md]: I am not sure if we should specify this here, as the current implementation differ in this informations for c++ and cpp clients.

**Review by bourtemb — COMMENTED** [review]
Fixed some typos

**Comment by piogor** [comment]
@bourtemb and @hardion, I think this is ready to be reviewed and then merged into the 'raw' branch :-)

**Pull request closed without merging**